### PR TITLE
Implementing unique id for deposits.

### PR DESCRIPTION
**Pull request by yabafinet**

### CODE REVIEW

File: src/Parsers/BHDBankParser.php [hunk 1]
```diff
@@ -34,12 +33,22 @@ public function parse(AbstractFile $file)
...
 
         return $collection;
     }
+
+    /**
+     * @param array $data
+     *
+     * @return string
+     */
+    public function uniqueId(array $data): string
```
Comment (maxalmonte14) [inline src/Parsers/BHDBankParser.php]: If does not exist in the interface, then doesn't have to exist in here.

```suggestion

```

File: src/Parsers/PopularBankParser.php [hunk 2]
```diff
@@ -31,12 +30,22 @@ public function parse(AbstractFile $file)
...
 
         return $collection;
     }
+
+    /**
+     * @param array $data
+     *
+     * @return string
+     */
+    public function uniqueId(array $data): string
```
Comment (maxalmonte14) [inline src/Parsers/PopularBankParser.php]: If does not exist in the interface, then doesn't have to exist in here.

```suggestion

```

File: src/Parsers/ReservasBankParser.php [hunk 3]
```diff
@@ -34,12 +33,22 @@ public function parse(AbstractFile $file)
...
 
         return $collection;
     }
+
+    /**
+     * @param array $data
+     *
+     * @return string
+     */
+    public function uniqueId(array $data): string
```
Comment (maxalmonte14) [inline src/Parsers/ReservasBankParser.php]: If does not exist in the interface, then doesn't have to exist in here.

```suggestion

```

File: src/Parsers/SantaCruzBankParser.php [hunk 4]
```diff
@@ -31,12 +30,22 @@ public function parse(AbstractFile $file)
...
 
         return $collection;
     }
+
+    /**
+     * @param array $data
+     *
+     * @return string
+     */
+    public function uniqueId(array $data): string
```
Comment (maxalmonte14) [inline src/Parsers/SantaCruzBankParser.php]: If does not exist in the interface, then doesn't have to exist in here.

```suggestion

```

File: src/Parsers/PopularBankParser.php [hunk 5]
```diff
@@ -31,12 +30,22 @@ public function parse(AbstractFile $file)
                 return;
             }
 
-            $collection->push(new Deposit($line[2], $line[0], $line[5], $line[1]));
+            $collection->push(new Deposit($line[2], $line[0], $line[5], $line[1], $this->uniqueId($line)));
```
Comment (maxalmonte14) [inline src/Parsers/PopularBankParser.php]: If every object has a unique id then it is not necessary to pass it via constructor.

```suggestion
$collection->push(new Deposit($line[2], $line[0], $line[5], $line[1]));
```

File: src/Deposit.php [hunk 6]
```diff
@@ -32,20 +32,29 @@ class Deposit
      */
     private $term;
 
+    /**
+     * The Unique Id the deposit.
+     *
+     * @var string
+     */
+    private $unique_id = null;
```
Comment (maxalmonte14) [inline src/Deposit.php]: This is null by default.

```suggestion
    private $uniqueId;
```

File: src/Collections/DepositCollection.php [hunk 7]
```diff
@@ -17,6 +18,10 @@ public function __construct($items = [])
             if (!$value instanceof Deposit) {
                 throw new InvalidArgumentException('You should pass only Deposit objects to this collection.');
             }
+
+            if (!$value->getUniqueId()) {
```
Comment (maxalmonte14) [inline src/Collections/DepositCollection.php]: If every object has a unique id this is not necessary.

```suggestion

```

**Review by maxalmonte14 — COMMENTED** [review]
I think we can generate a unique id in every `Deposit` object by default, that way we don't have to change other classes or interfaces or even pass the unique id to the class via constructor, if the unique id is for comparison purposes then every `Deposit` object should have one.

**Comment by yabafinet** [comment]
Implementing in another place.

**Pull request closed without merging**